### DISCUSSION TITLE
Good Looking Cases with sources (towards the end)

### DISCUSSION THREAD

**Discussion by alidaf** [sle118/squeezelite-esp32#287] [Show and tell]
This is my first ESP32 build and while it is early doors, I'm having some strange issues with playback & control.

When trying to pause in LMS, it immediately begins playing again, from the start of the track.
Album art doesn't always appear and doesn't change when a new track is selected in LMS.
Display doesn't update so time elapsed/remaining doesn't change.

Any thoughts?

**Comment by sle118** [comment]
without proper logs, this is going to be nearly impossible to to answer. Also, this can depend on where you are playing the music from (local music file, or using spotty?), network usage, server capacity, etc. A display that doesn't update could be a symptom of a hardware issue which causes the display not to receive refreshes. It could also indicate network contention, as menus/texts/elapsed, etc are always rendered on the server itself and pushed down through the network.

I'd monitor the network and most importantly I'd check for anything suspicious in the logs, captured from the serial port rather than from a telnet connection

**Reply by alidaf** [reply]
I've disabled Spotify and Airplay. I'm streaming directly from LMS using locally stored files.
I have other non-ESP32 squeeze players that work fine.
It may be because I have no other output hardware connected on this one yet, i.e. DAC - I'm still waiting for a delivery of some cheap 5102 boards to play with. 
I am concerned that this project will ultimately not be audiophile enough for my own use because of the limited hardware options unless I stick to only SPDIF out and rely on 3rd party or external DACs. I don't have the skills to design circuits, or the resources to have boards made but I will try to do enough to accommodate those that want to use their own ESP32-based hardware in my case, which I'll probably upload to [Printables](https://www.printables.com/) when I'm happy.
On the larger displays with Squeezelite-ESP32, menu navigation appears to be limited to a single line at the top of the display instead of making use of the entire area. It's very difficult to see, especially from a distance so I may just substitute for an SSD1322 and design another front of the case. It will be much more like my original Squeezebox, which I loved, and miss after mine broke.
I expect I'll have a number of different fronts and rears to accommodate all of the permutations available.
I'll keep at it, and thank you.

**Comment by philippe44** [comment]
If you're really into audiophile matters, this project is probably not the best fit for your needs.

**Reply by alidaf** [reply]
Forgive me if I appeared dismissive. I was referring to the original vision of my own project and the availability of 3rd party hardware rather than the potential of Squeezelite-esp32. Despite the ubiquity of the RPi and it's hardware options, I think Squeezelite-esp32 has the best potential for expansion and longevity. I'm re-evaluating my goals and squeezelite-esp32 will factor in them more prominently. As far as hardware is concerned, there is enough provision via SPDIF at the very least for a variety of alternatives. I was just very frustrated that the only DAC I could get was a very generic PCM5102 board from Aliexpress!

**Reply by philippe44** [reply]
Fair enough, I just want to be clear about my mindset as a dev. I'm also an engineer specialized in signal processing / information theory by education so I have some pretty defined opinions there. Now, I'm not on any crusade to convince people :smile: and anybody can hold any opinions on these topics, I just don't want to spend time debating that and even less coding some of it.

Having said that, did you make process on your issue?

**Reply by alidaf** [reply]
It just seemed to start working, thanks. I noted that there was an update but haven't read the patch notes so don't know if it had anything to do with it.
Also, I received another PCM5102 board (with RCA connectors as well as a headphone socket) and I was very pleasantly surprised, so that's going into a case for another room. The first board I bought only has a headphone socket and wasn't very good. I also received a SSD1322 display and prefer it over the ST7789 as it is much closer to the original Squeezebox VFD display so I'm designing a new case front for it. I bought one of these years ago and it cost a fortune but this was much more reasonable although everything from Aliexpress seems to be a bit of a gamble.
Given that the ESP32 components take up less space than the RPi I may try and design another case that much more closely resembles the original Squeezebox design.

**Reply by sle118** [reply]
@alidaf I hope you share the new designs here when they are done.

**Reply by alidaf** [reply]
@sle118 I will. This one is a bit of a hack though because I wanted it to be part of a suite with everything based on the same footprint and fascia size. The SSD1322 just about fits with enough room for the IR but no room for the rotary encoder or power button. I'll release it rough and ready and then start working on version 2. I just need to finish off the internal trays and back plate but these are specific to my internals. If I procure any other hardware I can work on adding options for those in the future, or just let the community add them. I think I'll prefer this one over my first design with the ST7789 because the SSD1322 is viewable from further. Being able to see album art is great 'n all but is only really of benefit to a more personal device (headphone amplifier) or a much bigger display. I'll probably end up just getting one of the A1 boards out of interest at some point. It's a little slow going at the moment but hopefully soon.

Here's the work in progress...

![Squeezelite-esp2](https://github.com/sle118/squeezelite-esp32/assets/14184640/153375d6-7824-4d00-b7e4-340be1d0a9e5)

**Comment by dfrein** [comment]
That looks quite nice, good luck with your project! I assume that you are aware that many (or even most) raspi dac hats can also be used with squeezelite-esp32 (that's what I did), so you are not limited to PCM5102 boards.

**Reply by alidaf** [reply]
Thanks. There's a different version with an ST7789 display elsewhere in the forums. I did read a post somewhere that at least one RPi hat was made to work and I have an old IQAudio hat that I was going to try and use at some point. I think some of the more elaborate boards such as those from ALLO use a lot of pins for other functions but don't provide much info on what, which makes it difficult to implement, and an expensive experiment. I'll be gearing up for it once I've finished and published the files for version 1.

**Comment by dfrein** [comment]
I would assume that most of the HATs without GPIOs reserved for configuration should work and will not require further init strings. For hifiberry devices you will find more information here: [hifiberry documentation](https://www.hifiberry.com/docs/hardware/gpio-usage-of-hifiberry-boards/)

**Reply by alidaf** [reply]
Thanks.

**Comment by alidaf** [comment]
@sle118 @philippe44 @dfrein 
Finally. https://www.printables.com/model/589907-squeezeboxen-case-v1

**Reply by sle118** [reply]
very nice!

**Reply by philippe44** [reply]
Would consider doing a SqueezeAMP version? I would appreciate it very much

**Reply by alidaf** [reply]
@philippe44 I've just finished a stand-alone class D amp based on a MAX9744 board (cheapish Adafruit one), which I like, but I'd like to give an integrated option a go and would love to have a go at the SqueezeAMP. Are there any fully populated boards available for order so I can give it a go? I'd order 2 for redundancy because I have a bad habit of knackering things during prototyping! It would probably have to be remote only - no room for a rotary unless I made it a bit bigger. I'm currently working on some more serious speakers with some Mark Audio drivers - the one's in the photo below are based on drivers from an unknown source (probably old desktop PC speakers) and not very good. They were more for proof of concept but the new ones will be much bigger and better quality since I can design around the known driver parameters.

![DSC_1636](https://github.com/sle118/squeezelite-esp32/assets/14184640/dea36d51-d074-4fa2-8c42-868fbf0bc502)

What kind of connections are you after? Spring loaded speaker terminals or proper banana plug posts?

BTW I much prefer the SSD1322 display and have ordered another one to try a hardware SPDIF option. I've completely gone off the PiCorePlayer version except for a LMS box, which will keep the ST7789 display - Squeezelite-ESP32 rules!

**Reply by philippe44** [reply]
I really do like the family! I have received a batch of SqueezeAMP so I have a few available. For the connectors, I really don't know, I would rather let you decide. The knob is not required but its true that 1 knob + 1 button is a better UX, although I see how it is impossible to fit with SSD1322 (this is the 2.8', not the 3.2', right?)

**Reply by philippe44** [reply]
oh, I just saw you're using the 3.2 display.  Maybe this [one](https://www.buydisplay.com/white-2-8-inch-arduino-raspberry-pi-oled-display-module-256x64-spi) would make a good solution as well and let the option of rotary and a button or two.

**Reply by alidaf** [reply]
Ok, I'll have a look at the smaller display. I'll have to order from AliExpress though, which might make the actual fit of your specific display need a bit of iteration for tolerances, and I'm not sure what to do about buttons just yet. I may make the boxes slightly bigger and take on board some of the things I learned with version 1 (e.g. use fewer heat set inserts). First things first though, how do I order the SqueezeAMPs?

When you say 1 knob + 1 button, does that account for the button on the rotary or do you prefer additional buttons? Does it matter if they are aligned horizontally or vertically for more intuitive use?

**Reply by philippe44** [reply]
I might have of these displays handy and if I have I can ship it to you with a SqueezeAMP. Maybe you can PM me so that I can get shipping address and all details.

**Reply by alidaf** [reply]
PM'ed on the Slim Devices forum.

**Reply by philippe44** [reply]
@alidaf - hey Darren, I've tried to answer to your email multiple times but I think your spam filter has decided to ignore me 😃

**Reply by alidaf** [reply]
@philippe44 I really don't know what's going on. It's not the spam filter - any spam ends up in a folder but your emails aren't even getting through to my MTA. I've been really panicked so used a Gmail address, but even those emails haven't had replies. I've sent some from my main email and then 5 from my Gmail account that have had step files but heard nothing from you since 27th November. I've checked my firewall in case you're IP is blacklisted but the previous emails that I have received are clear.
